eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -42,21 +42,16 @@ steps:
         ${{ parameters.SubscriptionConfiguration }}
       '@ | ConvertFrom-Json -AsHashtable;
 
-      if (!$subscriptionConfiguration.ArmTemplateParameters) {
-        $subscriptionConfiguration.ArmTemplateParameters = @{}
-      }
-
-      $armTemplateParameters = ${{ parameters.ArmTemplateParameters }}
-      foreach ($p in $armTemplateParameters.GetEnumerator()) {
-        $subscriptionConfiguration.ArmTemplateParameters[$p.Name] = $p.Value
-      }
-
       eng/common/TestResources/New-TestResources.ps1 `
         -BaseName 'Generated' `
         -ServiceDirectory '${{ parameters.ServiceDirectory }}' `
         -Location '${{ parameters.Location }}' `
         -DeleteAfterHours '${{ parameters.DeleteAfterHours }}' `
+        # The subscriptionConfiguration may have ArmTemplateParameters defined, so
+        # pass those in via the ArmTemplateParameters flag, and handle any
+        # additional parameters from the pipelines via AdditionalParameters
         @subscriptionConfiguration `
+        -AdditionalParameters ${{ parameters.ArmTemplateParameters }} `
```

Comment:
You said "pass those via the ArmTemplateParameters flag" (technically "parameter"), but then you pass to `AdditionalParameters`. Either the comment doesn't line up with the change, or I'm just confused.

Reply:
It's confusing. subscriptionConfiguration has an ArmTemplateParameters key, so we set that param (which is a new param as of this PR) when we splat @subscriptionConfiguration. But we still want to pass in any ones defined by the pipeline, and merge them all together in the script. So it leverages AdditionalParameters as a way to do that.

Reply:
It would be slightly easier to understand what's going on if we merged it here, but we're trying to avoid putting any more logic than necessary in the pipeline templates.

Reply:
Ah, okay, yes. I see it now. Thanks for clarifying!